### PR TITLE
[GD-109] Rework swithing network

### PR DESCRIPTION
**Pull request by sidorovkirill**

### CODE REVIEW

File: Assets/AnkrSDK/Scripts/Mobile/AnkrNetworkHelper.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+			var url = GetURLFromNetwork(network);
+			AddAndSwitchCustomNetwork(url);
+			
+			return Task.CompletedTask;
+		}
+
+		private static void AddAndSwitchCustomNetwork(string url)
+		{
+			Debug.Log(url);
+//			Application.OpenURL(url);
```
Comment (Vukiz) [inline Assets/AnkrSDK/Scripts/Mobile/AnkrNetworkHelper.cs:28]: meaningless comments are not allowed, also this method does nothing, but it should, right?

**Review by Vukiz — CHANGES_REQUESTED** [review]

File: Assets/AnkrSDK/Examples/UseCases/AddSwitchNetwork/EthereumNetworks.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+						Decimals = 18
+					},
+					RpcUrls = new[] {"https://data-seed-prebsc-1-s1.binance.org:8545/"},
+					BlockExplorerUrls = new[] {"https://testnet.bscscan.com"},
+					IconUrls = new[] {"future"}
+				}
+			}
+		};
+
+		public static EthereumNetwork GetSimpleNetwork(int chainId)
```
[the remark this answers is not in the document]
Comment (sidorovkirill) [reply Assets/AnkrSDK/Examples/UseCases/AddSwitchNetwork/EthereumNetworks.cs:51]: Cahnged to CreateMetamaskExistedNetwork

File: Assets/AnkrSDK/Scripts/Data/EthereumNetwork.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+		public string[] RpcUrls { get; set; }
+		[CanBeNull]
+		[JsonProperty("blockExplorerUrls")]
+		public string[] BlockExplorerUrls { get; set; }
+		[CanBeNull]
+		[JsonProperty("iconUrls")]
+		public string[] IconUrls { get; set; }
+	}
+
+	public class NativeCurrency
```
[the remark this answers is not in the document]
Comment (sidorovkirill) [reply Assets/AnkrSDK/Scripts/Data/EthereumNetwork.cs:26]: This class don't need without EthereumNetwork

File: Assets/AnkrSDK/Scripts/Data/NetworkName.cs [hunk 4]
```diff
@@ -3,9 +3,10 @@ namespace AnkrSDK.Data
 	public enum NetworkName
 	{
 		Ethereum,
-		Ethereum_Rinkeby_TestNet,
-		Ethereum_Goerli_TestNet,
-		Ethereum_Ropsten_TestNet,
```
[the remark this answers is not in the document]
Comment (sidorovkirill) [reply Assets/AnkrSDK/Scripts/Data/NetworkName.cs:8]: Of course it's a testnets, but those names well-known, and otherwice we have test nets that duplicate original nets. And in that case I think we need to mark it like _TestNet. But not in the case of well-known test networks.

File: Assets/AnkrSDK/Scripts/Data/EthereumNetwork.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (Vukiz) [reply Assets/AnkrSDK/Scripts/Data/EthereumNetwork.cs:23]: Does not matter if it is public it shouldn't be inside file with another class

File: Assets/AnkrSDK/Scripts/Mobile/AnkrNetworkHelper.cs [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+		public Task AddAndSwitchNetwork(EthereumNetwork network)
+		{
+			var url = GetURLFromNetwork(network);
+			AddAndSwitchCustomNetwork(url);
+			
+			return Task.CompletedTask;
+		}
+
+		private static void AddAndSwitchCustomNetwork(string url)
```
Comment (Vukiz) [inline Assets/AnkrSDK/Scripts/Mobile/AnkrNetworkHelper.cs:25]: don't make it private

**Review by Vukiz — CHANGES_REQUESTED** [review]

File: .idea/.idea.game-unity-sdk/.idea/indexLayout.xml [hunk 6]
```diff
@@ -1,5 +1,10 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
+  <component name="ContentModelUserStore">
```
[the remark this answers is not in the document]
Comment (sidorovkirill) [reply .idea/.idea.game-unity-sdk/.idea/indexLayout.xml:3]: It's already there

[the remark this answers is not in the document]
Comment (Vukiz) [reply .idea/.idea.game-unity-sdk/.idea/indexLayout.xml:3]: Well then you should remove it from index then

**Review by Vukiz — APPROVED** [review]

**Pull request merged**